labs/tasks/repository.py

```diff
@@ -1,32 +1,67 @@
 import json
-import logging
+from types import SimpleNamespace
 
 import config.configuration_variables as settings
-import redis
 from config.celery import app
-from repo import call_agent_to_apply_code_changes, clone_repository
-from run import commit_changes, create_branch, create_pull_request, get_issue
+from decorators import time_and_log_function
+from github.github import GithubRequests
+from parsers.response_parser import create_file, modify_file, parse_llm_output
+from tasks.redis_client import RedisStrictClient, RedisVariables
+
+redis_client = RedisStrictClient(host=settings.REDIS_HOST, port=settings.REDIS_PORT, db=0, decode_responses=True)
+
+
+def get_redis_github_data(
+    prefix, token="", repository_owner="", repository_name="", username="", as_namespace=False
+) -> dict | SimpleNamespace:
+    variables = dict(
+        token=redis_client.get(RedisVariables.TOKEN, prefix, default=token),
+        repository_owner=redis_client.get(RedisVariables.REPOSITORY_OWNER, prefix, default=repository_owner),
+        repository_name=redis_client.get(RedisVariables.REPOSITORY_NAME, prefix, default=repository_name),
+        username=redis_client.get(RedisVariables.USERNAME, prefix, default=username),
+    )
+
+    if as_namespace:
+        return SimpleNamespace(**variables)
+
+    return variables
+
+
+@time_and_log_function
+def apply_code_changes(llm_response):
+    pull_request = parse_llm_output(llm_response)
 
-logger = logging.getLogger(__name__)
+    files = []
+    for step in pull_request.steps:
```

Comment:
nit: `pull_request` should be renamed to whatever you named it on `response_parser.py`